paddle/framework/op_registry.h

```diff
@@ -61,7 +61,15 @@ class OpProtoAndCheckerMaker {
   OpProtoAndCheckerMaker(OpProto* proto, OpAttrChecker* op_checker)
       : proto_(proto), op_checker_(op_checker) {}
 
-  ~OpProtoAndCheckerMaker() { CheckNoDuplicatedAttrs(); }
+  ~OpProtoAndCheckerMaker() {
+    PADDLE_ENFORCE(validated_, "should call Validate after build");
+  }
+
+  void Validate() {
+    validated_ = true;
+    CheckNoDuplicatedAttrs();
+    CheckNoDuplicatedInOut();
```

Comment:
Not only attributes but also input and output together could not be duplicated.

So maybe one function named `CheckNoDuplicatedInOutAttrs` is better?